### ISSUE TITLE
problem: incrementing  prereleases

### ISSUE THREAD

**Issue by xenoterracide** [semver4j/semver4j#243]
so, I've reviewed the semver spec. There's not much more than convention around pre-release strings. I mean it does have rules about how to handle them, but there's not a super strong structure there like there is with major.minor.patch. 

What I need is to be able to increment one

So if a user tags their git repo `v1.1.0-rc.1` and then they need to create a "release" of a commit on top of that version. Per build jar creation. I would need to increment to a version that's something like `v1.1.0-rc.2.alpha.1` or `v1.1.0-rc.2.1`

Helping on this might be out of the scope of this project, because I think it would require adding a convention on top of the semver spec.

The ask is for an api to do `nextPrerelease()` which would increment from rc.1 to rc.2 . That seemed to be where I was hitting complexity. Although I'm going to speculate that's partially do to #236 and #241 where I'm having to dig around and split, join, and convert strings to integers etc.

. I think that's the extent, but think over the problem maybe...

https://github.com/xenoterracide/gradle-semver/blob/main/src/main/java/com/xenoterracide/gradle/semver/SemverExtension.java#L68

**Comment by piotrooo** [comment]
I like the idea behind this proposal. However, there are many questions about how it should work in corner case scenarios. Specifically, how should we handle the following version strings:

* `1.1.1-rc1`
* `1.1.1-rc1.alpha.2`
* `1.1.1-rc.1.2`
* and many more

We need some kind of arbitrary logic to handle these cases.

**Comment by xenoterracide** [comment]
Yeah, I wonder if it's better to arbitrarily handle everything instead of coming up with a convention of what you think it should be... Meaning maybe don't handle incrementing for every arbitrary option. 

On the other hand I ended up deciding that it was better for me to use the commit distance as a sub subversion. Eg I always have rc.1.2 where the two would be The commit distance. This is because I figured people would want to tag things as rc1 themselves. Although my design is also specifically for doing snapshot releases in a way that's not snapshots...
